packages/block-editor/src/components/color-palette/with-color-context.js

```diff
@@ -10,14 +10,17 @@ import { useSettings } from '../use-settings';
 
 export default createHigherOrderComponent( ( WrappedComponent ) => {
 	return ( props ) => {
-		const [ colorsFeature, enableCustomColors ] = useSettings(
-			'color.palette',
+		// Get the default colors, theme colors, and custom colors
+		const [ themeColors, customColors, enableCustomColors ] = useSettings(
+			'color.palette.theme',
+			'color.palette.custom',
 			'color.custom'
 		);
-		const {
-			colors = colorsFeature,
-			disableCustomColors = ! enableCustomColors,
-		} = props;
+
+		const colors = [ ...( themeColors || [] ), ...( customColors || [] ) ];
```

Comment:
```suggestion
		const colors = [ ...themeColors, ...defaultColors, ...customColors ];
```

This simplification should be fine.